src/Internal/Calculator.php

```diff
@@ -57,10 +59,13 @@ abstract class Calculator
      * If none has been explicitly set, the fastest available implementation will be returned.
      *
      * @return Calculator
+     *
+     * @psalm-pure
```

Comment:
I'm a bit reluctant to call this function "pure": it does cause a side effect, which is to switch calculators.
Did you declare it as pure because it's used by factory methods, that we do want to be marked as pure?

How can this be solved in an elegant way?

Reply:
I marked it as pure because:
- it returns a calculator anyway, just the implementation can be different
- it doesn't change anything inside/outside

So even that the function checks for extensions, it's a kind of optimization, with no side effects for the end user.

Reply:
> with no side effects for the end user.

I guess this is what matters indeed.